pkg/skaffold/runner/deployer.go

```diff
@@ -86,13 +90,28 @@ func GetDeployer(runCtx *v2.RunContext, provider deploy.ComponentProvider, label
 			deployers = append(deployers, deployer)
 		}
 
+		// TODO: deprecate KustomizeDeploy
 		if d.KustomizeDeploy != nil {
 			deployer, err := kustomize.NewDeployer(dCtx, labels, provider, d.KustomizeDeploy)
 			if err != nil {
 				return nil, err
 			}
 			deployers = append(deployers, deployer)
 		}
+
+		if d.KptV2Deploy != nil {
+			if d.KptV2Deploy.Dir == "" {
+				logrus.Debugln("use default render directory as deployment directory.")
+				hydrationDir, err := runCtx.GetRenderOutputPath()
+				if err != nil {
+					return nil, err
+				}
+				logrus.Infof("manifests are deployed from render path %v\n", hydrationDir)
+				d.KptV2Deploy.Dir = hydrationDir
+			}
+			deployer := kptV2.NewDeployer(dCtx, labels, provider, d.KptV2Deploy)
```

Comment:
should we be handling this somewhere else in execution? We have logic for setting default field values in `pkg/skaffold/schema/defaults`

Reply:
Yeah, this is the PR about handling default behavior. https://github.com/GoogleContainerTools/skaffold/pull/6252/files
kptV2 deployer is not the default one (so far). We are still going to use kubectl. Here's the related [doc](https://docs.google.com/document/d/1fH450puJ7An_MReChT705j8uBO3w585gm34Tbm9IVWY/edit?resourcekey=0-xYq8VW3ZG7Ait5ag1nDNew#heading=h.jcx526m6nqg4) (See Iterm #2) 

Do you suggest some additional handling on the default.go?